### PR TITLE
Changed HttpHeaders to use Utf8String

### PR DESCRIPTION
**Pull request by livarcocc**
Modified HttpHeaders to use Utf8String instead of ByteSpan.

- It creates an Utf8String from the ByteSpan and manipulates that.
- The next step is to have a list of headers so that we can do operations like Add/Remove header.

### CODE REVIEW

File: demos/LowAllocationWebServer/HttpHeaders.cs [hunk 1]
```diff
@@ -67,60 +71,57 @@ IEnumerator IEnumerable.GetEnumerator()
...
-            int consumedBytes;
-            var headerName = new Utf8String(bytes.SliceTo(start, ':', out consumedBytes));
-            parsedBytes = start + consumedBytes;
+            Utf8String headerName;
+            Utf8String headerValue;
 
-            var headerValue = new Utf8String(bytes.SliceTo(start + consumedBytes, '\r', '\n', out consumedBytes));
-            parsedBytes += consumedBytes;
+            //TODO: this will be simplified once we have TrySubstringTo/From accepting strings
+            headerString.TrySubstringTo((Utf8CodeUnit) (byte) ':', out headerName);
```
Comment (KrzysztofCwalina) [inline demos/LowAllocationWebServer/HttpHeaders.cs]: What if the Try returns false?

File: demos/LowAllocationWebServer/HttpHeaders.cs [hunk 2]
```diff
@@ -67,60 +71,65 @@ IEnumerator IEnumerable.GetEnumerator()
...
+        private static Utf8String ParseHeaderLine(Utf8String headerString, out KeyValuePair<Utf8String, Utf8String> header)
         {
-            int consumedBytes;
-            var headerName = new Utf8String(bytes.SliceTo(start, ':', out consumedBytes));
-            parsedBytes = start + consumedBytes;
+            Utf8String headerName;
+            Utf8String headerValue;
+
+            //TODO: this will be simplified once we have TrySubstringTo/From accepting strings            
+            if (!headerString.TrySubstringTo((Utf8CodeUnit) (byte) ':', out headerName))
```
Comment (KrzysztofCwalina) [inline demos/LowAllocationWebServer/HttpHeaders.cs]: @krwq, shjould we add an overload that takes char?

**Comment by KrzysztofCwalina** [comment]
Looks good. Please squash the commits and merge. Thanks!

**Comment by livarcocc** [comment]
I hope I did the squash right.

**Comment by KrzysztofCwalina** [comment]
hmm, it still shows 15 commits.

**Pull request merged**